applications/web_app/rest/dotnet/ci/GitHub/.github/workflows/init_repository.yml.j2

```diff
@@ -0,0 +1,27 @@
+name: Generate dotnet REST API
+on:
+  push:
+    branches:
+      - main
+jobs:
+  generate-template:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      
+      - uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: '6.0.x'
+      
+      - name: Generate template
+        {% if ctx.variables["total_app_templates"] > 1 -%}
+          working-directory: {{ctx.current_template['template_name']}}
+        {%- endif %}
+        run: dotnet new webapi {{ctx.current_template['parameters']['DOTNET_API_TYPE']}} -o rest
+      
+      - name: Remove initialization workflow
+        run: rm .github/workflows/init_repository.yml
+      
+      - uses: stefanzweifel/git-auto-commit-action@v4
```

Comment:
Does this actually work if branch protection rules are enforced (PR required to main with at least 1 review)? These are set by sprout if the repository is in an organisation

Reply:
IE if the action cannot push the changes to main it will create a branch out of it and create a PR to main?

Reply:
https://github.com/stefanzweifel/git-auto-commit-action/issues/87 Looks like it won't, not without a PAT at least. ☹️